### PR TITLE
Introduce dedicated registration test

### PR DESCRIPTION
**Pull request by pdostal**
- Related ticket: [poo#138221](https://progress.opensuse.org/issues/138221)
- Related merge request: [gitlab.suse.de/qac/qac-openqa-yaml!1416](https://gitlab.suse.de/qac/qac-openqa-yaml/-/merge_requests/1416)
- Verification run: [aggregates](https://pdostal-server.suse.cz/tests/5441), [latest](https://pdostal-server.suse.cz/tests/5442), [registrations](https://pdostal-server.suse.cz/tests/5440)

### CODE REVIEW

**Review by ricardobranco777 — APPROVED** [review]

**Comment by asmorodskyi** [comment]
please also change same flow for latest

**Comment by pdostal** [comment]
> please also change same flow for latest

Done.

File: lib/main_publiccloud.pm [hunk 1]
```diff
@@ -120,17 +120,18 @@ sub load_latest_publiccloud_tests {
     elsif (&$should_use_runargs()) {
         my $args = OpenQA::Test::RunArgs->new();
         loadtest "publiccloud/prepare_instance", run_args => $args;
-        if (get_var('PUBLIC_CLOUD_CONSOLE_TESTS') && !is_container_host()) {
-            loadtest("publiccloud/check_registercloudguest", run_args => $args);
+        if (get_var('PUBLIC_CLOUD_REGISTRATION_TESTS')) {
```
Comment (asmorodskyi) [inline lib/main_publiccloud.pm:123]: why you removing it for loadtest but keep it for get_var ? :thinking:

**Comment by asmorodskyi** [comment]
I still can merge your PR as is , but in such a case whenever I will come up with PR for this file I will re-enter all brackets which you have deleted and also will add them everywhere else :D

File: lib/main_publiccloud.pm [hunk 2]
```diff
@@ -120,17 +120,18 @@ sub load_latest_publiccloud_tests {
...
+            loadtest "publiccloud/check_registercloudguest", run_args => $args;
```
[the remark this answers is not in the document]
Comment (pdostal) [reply lib/main_publiccloud.pm]: Reverted.

File: lib/main_publiccloud.pm [hunk 3]
```diff
@@ -24,12 +24,7 @@ sub load_maintenance_publiccloud_tests {
 
     loadtest "publiccloud/download_repos";
     loadtest "publiccloud/prepare_instance", run_args => $args;
-    if (get_var('PUBLIC_CLOUD_CONSOLE_TESTS')) {
-        loadtest("publiccloud/check_registercloudguest", run_args => $args);
-    }
-    else {
-        loadtest("publiccloud/registration", run_args => $args);
-    }
+    loadtest("publiccloud/registration", run_args => $args);
```
Comment (asmorodskyi) [inline lib/main_publiccloud.pm:27]:
```suggestion
    loadtest("publiccloud/registration", run_args => $args) unless get_var('PUBLIC_CLOUD_REGISTRATION_TESTS');
```

:thinking:  ?

Comment (asmorodskyi) [reply lib/main_publiccloud.pm:27]: isn't  `check_registercloudguest` expect to start on unregistered system ? :thinking:

Comment (asmorodskyi) [reply lib/main_publiccloud.pm:27]: ah sorry , all this code collapsing confused me . please ignore this message

**Review by asmorodskyi — APPROVED** [review]
LGTM

**Pull request merged**